neo/Ledger/Blockchain.cs

```diff
@@ -253,6 +255,27 @@ private void AddUnverifiedBlockToCache(Block block)
             blocks.AddLast(block);
         }
 
+        private void OnFillMemoryPool(IEnumerable<Transaction> transactions)
+        {
+            // Invalidate all the transactions in the memory pool, to avoid any failures when adding new transactions.
+            MemPool.InvalidateAllTransactions();
+
+            // Add the transactions to the memory pool
+            foreach (var tx in transactions)
+            {
+                // First remove the tx if it is unverified in the pool.
+                MemPool.TryRemoveUnVerified(tx.Hash, out _);
+                // Verify the the transaction
+                if (!tx.Verify(currentSnapshot, MemPool.GetVerifiedTransactions()))
```

Comment:
https://github.com/neo-project/neo/blob/ccb1d70c930f5a54498cceef70442a1edca1a86c/neo/Ledger/Blockchain.cs#L393-L411

We should do these checks on the transactions here.

Reply:
I added the checks that apply.